apps/passport-client/components/screens/HomeScreen/HomeScreen.tsx

```diff
@@ -50,32 +51,30 @@ export function HomeScreenImpl(): JSX.Element | null {
   const self = useSelf();
   const navigate = useNavigate();
   const dispatch = useDispatch();
+  const loadedIssuedPCDs = useLoadedIssuedPCDs();
 
-  const pcdCollection = usePCDCollection();
   const [searchParams, setSearchParams] = useSearchParams();
   const defaultBrowsingFolder = useMemo(() => {
     const folderPathFromQuery = decodeURIComponent(
       searchParams.get(FOLDER_QUERY_PARAM) ?? ""
     );
-    if (!folderPathFromQuery) {
+    if (!folderPathFromQuery || folderPathFromQuery === "null") {
       return "";
     }
     // FrogCrypto is always valid even if user doesn't have any FrogPCD
     if (folderPathFromQuery === FrogCryptoFolderName) {
       return folderPathFromQuery;
     }
 
-    return pcdCollection.isValidFolder(folderPathFromQuery)
-      ? folderPathFromQuery
-      : "";
-  }, [pcdCollection, searchParams]);
+    return folderPathFromQuery || "";
+  }, [searchParams]);
 
   const [browsingFolder, setBrowsingFolder] = useState(defaultBrowsingFolder);
   const pcdsInFolder = useVisiblePCDsInFolder(browsingFolder);
   const foldersInFolder = useFolders(browsingFolder);
 
   useEffect(() => {
-    if (!self) {
+    if (!self && !browsingFolder) {
```

Comment:
this feels kind of like a hack - is there a way for `self` to be defined by the time you get here?